firebase-common/src/main/java/com/google/firebase/heartbeatinfo/HeartBeatInfoStorage.java

```diff
@@ -127,10 +127,18 @@ synchronized void clearStoredHeartBeats() {
     sharedPreferences.edit().remove(HEART_BEAT_COUNT_TAG).apply();
   }
 
-  static boolean isSameDateUtc(long base, long target) {
+  synchronized static boolean isSameDateUtc(long base, long target) {
     Date baseDate = new Date(base);
     Date targetDate = new Date(target);
-    return !(FORMATTER.format(baseDate).equals(FORMATTER.format(targetDate)));
+    String baseDateString = baseDate
+            .toInstant()
```

Comment:
`toInstant()` is not available in Java < 8 so can only be used on devices with API < 24/

Reply:
Removed it